typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +150,541 @@ export class EvmIsmModule extends HyperlaneModule<
+  }): Promise<DomainRoutingIsm> {
+    const {
+      destination,
+      owner,
+      safeConfigDomains,
+      submoduleAddresses,
+      overrides,
+      multiProvider,
+      factories,
+    } = params;
+
+    // deploying new domain routing ISM
+    const tx = await factories.domainRoutingIsmFactory.deploy(
+      owner,
+      safeConfigDomains,
+      submoduleAddresses,
+      overrides,
+    );
+
+    const receipt = await multiProvider.handleTx(destination, tx);
+
+    // TODO: Break this out into a generalized function
+    const dispatchLogs = receipt.logs
+      .map((log) => {
+        try {
+          return factories.domainRoutingIsmFactory.interface.parseLog(log);
+        } catch (e) {
+          return undefined;
+        }
+      })
+      .filter(
+        (log): log is ethers.utils.LogDescription =>
+          !!log && log.name === 'ModuleDeployed',
+      );
+    if (dispatchLogs.length === 0) {
+      throw new Error('No ModuleDeployed event found');
+    }
+    const moduleAddress = dispatchLogs[0].args['module'];
+    return DomainRoutingIsm__factory.connect(
+      moduleAddress,
+      multiProvider.getSigner(destination),
+    );
+  }
+
+  protected static async deployAggregationIsm(params: {
+    destination: ChainName;
+    config: AggregationIsmConfig;
+    origin?: ChainName;
+    mailbox?: Address;
+    logger: Logger;
+    multiProvider: MultiProvider;
+    factories: HyperlaneContracts<ProxyFactoryFactories>;
+    deployer: HyperlaneDeployer<any, any>;
+  }): Promise<IAggregationIsm> {
+    const {
+      destination,
+      config,
+      origin,
+      mailbox,
+      logger,
+      multiProvider,
+      factories,
+      deployer,
+    } = params;
+    const signer = multiProvider.getSigner(destination);
+    const staticAggregationIsmFactory = factories.staticAggregationIsmFactory;
+    const addresses: Address[] = [];
+    for (const module of config.modules) {
+      const submodule = await EvmIsmModule.deploy({
+        destination,
+        config: module,
+        origin,
+        mailbox,
+        multiProvider,
+        factories,
+        deployer,
+      });
+      addresses.push(submodule.address);
+    }
+    const address = await EvmIsmModule.deployStaticAddressSet({
+      chain: destination,
+      factory: staticAggregationIsmFactory,
+      values: addresses,
+      logger: logger,
+      threshold: config.threshold,
+      multiProvider,
+    });
+    return IAggregationIsm__factory.connect(address, signer);
+  }
+
+  protected static async deployStaticAddressSet(params: {
+    chain: ChainName;
+    factory: StaticThresholdAddressSetFactory | StaticAddressSetFactory;
+    values: Address[];
+    logger: Logger;
+    threshold?: number;
+    multiProvider: MultiProvider;
+  }): Promise<Address> {
+    const {
+      chain,
+      factory,
+      values,
+      logger,
+      threshold = values.length,
+      multiProvider,
+    } = params;
+    const sorted = [...values].sort();
```

Comment:
Well done! I see people make the mistake of sorting the original array very often